index2.js

```diff
@@ -0,0 +1,269 @@
+12345`,
+  false,
+  "That would be selfish. How will other students find their way?",
+  "",
+  "",
+);
+
+const door = new Item(
+  "door",
+  "182 Main St",
+  false,
+  "It's locked and I don't know how you'd carry it.",
+  "",
+  "The door is locked. There is a keypad on the door handle."
+);
+
+//Rooms
+let mainSt = new Room(
+  "182 Main St.",
+  `You are standing on Main Street between Church and South Winooski.
+There is a door here. A keypad sits on the handle.
+On the door is a handwritten sign.`,
+  [sign, door],
+  ['mainSt', 'foyer']
+  );
+
+let foyer = new Room(
+'182 Main St. - Foyer',
+`You are in a foyer. Or maybe it's an antechamber. Or a 
+vestibule. Or an entryway. Or an atrium. Or a narthex.
+But let's forget all that fancy flatlander vocabulary,
+and just call it a foyer. In Vermont, this is pronounced
+"FO-ee-yurr".
+A copy of Seven Days lies in a corner.`,
+[paper],
+['street', 'roomThree'],
+);
+
+  //Do we need a lookup table for [current].description to work? Marshall mentioned that you can't have the first 
+  //key be a variable unless you use a lookup table to convert the strimovng to an object by naming itself
+  const roomLookupTable = {
+    mainSt: mainSt,
+    foyer: foyer,
+    "street": mainSt
+    //roomThree: roomThree
+  }
+  
+  let player = new Player(
+    'street',
+    []
+    );
+  
+  const lookUpItems = {
+     paper: paper,
+     "seven days": paper,
+     sign: sign,
+     door: door,
+     }
+
+//Transition between rooms
+function enterState(newState, currentRoom) {
+   // console.log(roomLookupTable[currentRoom].canChangeTo)
+  // console.log(`Entering enterState() the currenState is: ${roomLookupTable[currentRoom].canChangeTo}`);
+  // console.log(`enterState with ${newState} as the argument`);
+  if(roomLookupTable[currentRoom].canChangeTo.includes(newState)){
+    player.currentState = newState;
+    console.log(roomLookupTable[player.currentState].name);
+    console.log(roomLookupTable[player.currentState].description);
+    getInput();
+  }else{
+    throw 'Invalid state transition attempted - from ' + currentRoom + ' to ' + newState;
+  }
+}
+
+function start() {
+  console.log("Please enter your commands as two words. For example 'open door' or 'read sign'.");
+  player.currentState = 'mainSt';
+  enterState('mainSt', 'mainSt');
+}
+
+async function getInput() {
+  let input = await ask("\nWhat would you like to do?\n>_");
+  let arrInput = input.toLowerCase().split(" ");
+  if(arrInput[1] === 'seven' && arrInput[2] === 'days'){
+    arrInput.push('seven days');
+  }
+  if(arrInput.includes('enter', 'code') || arrInput.includes('key', 'in')){
+    return player.code(arrInput[2]);
+  }
+  checkInput(arrInput[0], arrInput[arrInput.length-1], arrInput);
+}
+
+
+function checkInput(arg1, arg2, arrInput){
+  if(arg1 === 'i' || (arg1 === 'take' && arg2 === 'inventory')){
+      arg1 = 'inventory';
+      // if(player.playerInventory.length < 1){
```

Comment:
Once you are done with certain blocks of code by commenting them out, you can delete said blocks of code prior to submitting your project.